app/preload/database/transactionsDao.ts

```diff
@@ -80,6 +81,21 @@ async function hasEncryptedTransactions(
     return Boolean(transaction);
 }
 
+async function hasPendingShieldedBalanceTransfer(fromAddress: string) {
```

Comment:
You need to add some indices to support this query. Currently all it will use is the one on `fromAddress`, and for e.g. a baker account this will be insufficient to make this check effective. A reasonable assumption is that there will always be a limited amount of `Pending` transactions, so in this case you can probably just add an index for TransactionStatus. We will probably need an index on `transactionKind` at some point as well, so consider adding that now too.

Reply:
I'll add the indices.